flow/record/fieldtypes/__init__.py

```diff
@@ -734,3 +734,26 @@ def __repr__(self) -> str:
                 quote = '"'
 
         return f"{quote}{s}{quote}"
+
+
+# DIS-2557: str(Path("")) should not return "."
+if not hasattr(pathlib.PurePath, "__old_str__"):
+    if PY_312:
+
+        def _new__str__(self) -> str:
+            return "" if "".join(self._raw_paths).strip() == "" else pathlib.PurePath.__old_str__(self)
+
+    else:
+
+        def _new_parse_parts(self, parts: list[str | pathlib.PurePath]) -> tuple(str, str, str):
+            self.empty = "".join(parts).strip() == ""
+            return pathlib._Flavour._old_parse_parts(self, parts)
+
+        def _new__str__(self) -> str:
+            return "" if self._flavour.empty else pathlib.PurePath.__old_str__(self)
+
+        pathlib._Flavour._old_parse_parts = pathlib._Flavour.parse_parts
+        pathlib._Flavour.parse_parts = _new_parse_parts
+
+    pathlib.PurePath.__old_str__ = pathlib.PurePath.__str__
+    pathlib.PurePath.__str__ = _new__str__
```

Comment:
Wouldn't this change the behaviour of _all_ `Path` objects as a side effect?

```python
In [1]: from flow.record import fieldtypes

In [2]: from pathlib import Path

In [3]: Path("")
Out[3]: PosixPath('')
```

Reply:
Otherwise:

`path(Path(""))`

will still give you `"."`

Maybe I am missing something but to make this work I think I need to change the default behavior of Path, otherwise the data is lost. Or we have to replace every instance of Path with a different class, like ForensicPath (that keeps the original args)?

Reply:
I don't think we want to change default behaviour of stdlib classes.


> path(Path(""))
> 
> will still give you "."
> 

I think the above is "expected" behaviour as stdlib Path does not support empty paths, but we do want to support it in our own path objects.